cinema/serializers.py

```diff
@@ -1 +1,85 @@
-# write serializers here
+from rest_framework import serializers
+from cinema.models import Genre, Actor, CinemaHall, Movie, MovieSession
+
+
+class ActorSerializer(serializers.ModelSerializer):
+    full_name = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Actor
+        fields = ("id", "first_name", "last_name", "full_name")
+
+    def get_full_name(self, obj):
+        return f"{obj.first_name} {obj.last_name}"
+
+
+class GenreSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Genre
+        fields = ("id", "name")
+
+
+class BaseMovieSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Movie
+        fields = ("id", "title", "description", "duration", "genres", "actors")
+
+
+class MovieListSerializer(BaseMovieSerializer):
+    genres = serializers.SlugRelatedField(
+        many=True,
+        read_only=True,
+        slug_field="name"
+    )
+    actors = serializers.SerializerMethodField()
+
+    def get_actors(self, obj):
+        return [
+            f"{actor.first_name} {actor.last_name}"
+            for actor in obj.actors.all()
+        ]
+
+
+class MovieRetrieveSerializer(BaseMovieSerializer):
+    genres = GenreSerializer(many=True)
+    actors = ActorSerializer(many=True)
+
+
+class CinemaHallSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CinemaHall
+        fields = ("id", "name", "rows", "seats_in_row", "capacity")
+
+
+class MovieSessionSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = MovieSession
+        fields = ("id", "show_time", "movie", "cinema_hall")
+
+
+class MovieSessionListSerializer(MovieSessionSerializer):
+    movie_title = serializers.SlugRelatedField(
+        source="movie", slug_field="title", read_only=True
+    )
+    cinema_hall_name = serializers.SlugRelatedField(
+        source="cinema_hall", slug_field="name", read_only=True
+    )
+    cinema_hall_capacity = serializers.SlugRelatedField(
+        source="cinema_hall", slug_field="capacity", read_only=True
+    )
```

Comment:
Using `SlugRelatedField` for `cinema_hall_capacity` might not be appropriate if `capacity` is not a unique field. Consider using a different serializer field, such as `ReadOnlyField`, if `capacity` is not unique.